src/streamlink/plugins/discovery.py

```diff
@@ -0,0 +1,72 @@
+import re
+
+from urllib.parse import unquote
+from json import loads
+
+from streamlink.plugin import Plugin
+from streamlink.plugin import PluginError
+from streamlink.stream import HLSStream
+
+
+class Discovery(Plugin):
+    url_re = re.compile(r'''
+        ^https?://(www\.)?((go|watch)\.)?
+        (?P<host>
+            discovery|tlc|animalplanet|investigationdiscovery|science|discoverylife|
+            motortrend|ahc|destinationamerica|discoveryfamily|discoveryenespanol|
+            discovery-familia|own|hgtv|food|travel|cooking-channel|diy
+        )
+        \.com/watch/
+        (?P<slug>
+            discovery|tlc|animal-planet|investigation-discovery|science|discovery-life|
+            motortrend|ahc|destination-america|discovery-family|discovery-en-espanol|
+            discovery-familia|own|hgtv|food|travel|cooking-channel|diy
+        )
+        \??$
+    ''', re.VERBOSE)
+
+    api_url = 'https://api.discovery.com/v1/content/livestreams?slug='
+
+    @classmethod
+    def can_handle_url(cls, url):
+        return cls.url_re.match(url) is not None
+
+    def _get_streams(self):
+        slug = self.url_re.match(self.url)['slug'].lower()
```

Comment:
We need to maintain Python 2 compatibility, you need to use `groups()` as Python 2 regex match items don't support `__getitem__`.